test/e2e/helpers/broker_expression_test_helper.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helpers
+
+import (
+	"testing"
+
+	"knative.dev/eventing/pkg/reconciler/sugar"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	cetest "github.com/cloudevents/sdk-go/v2/test"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	testlib "knative.dev/eventing/test/lib"
+	"knative.dev/eventing/test/lib/recordevents"
+	"knative.dev/eventing/test/lib/resources"
+)
+
+// TestBrokerWithExpressionTrigger If shouldLabelNamespace is set to true this test annotates the testing namespace so that a default broker is created.
+// It then binds many triggers with different filtering patterns to the broker created by brokerCreator, and sends
+// different events to the broker's address.
+// Finally, it verifies that only the appropriate events are routed to the subscribers.
+func TestBrokerWithExpressionTrigger(t *testing.T, brokerCreator BrokerCreator, shouldLabelNamespace bool) {
```

Comment:
Is the label namespace handling really necessary?

Reply:
As before, no clue, i took that code from broker with many triggers test :smile:

Reply:
I'm pretty sure we don't need it in this test.